sparse/coo/common.py

```diff
@@ -212,6 +212,65 @@ def _dot(a, b):
     return aa.dot(b)
 
 
+def kron(a, b):
```

Comment:
What is the approach in sparse when coercing inputs? If neither input is a sparse array, do you return a dense result, or coerce to sparse?

Reply:
We generally raise a `ValueError` if the result is guaranteed to be dense.

Reply:
The `dot` behaviour is weird, it should work on mixed or sparse only, but not on dense-dense. To answer your question: In general, we only accept dense inputs if mixed sparse-dense operations are more efficient than a dense-only one. If the dense-only operation is the best one available, we err.

For `dot`, there is an efficient implementation of sparse-dense and dense-sparse multiplication, as in this case. For element-wise operations, this is sometimes the case (think `operator.mul`), but not always (think `operator.add`).

This case is similar: A mixed operation could benefit from the sparse structure, but not a dense-only one.